report_xlsx/report/report_xlsx.py

```diff
@@ -24,11 +24,15 @@ def create(self, cr, uid, ids, data, context=None):
         if report.ids:
             self.title = report.name
             if report.report_type == 'xlsx':
-                objs = self.env[self.table].browse(ids)
-                return self.create_xlsx_report(data, objs)
+                return self.create_xlsx_report(ids, data, report)
         return super(ReportXlsx, self).create(cr, uid, ids, data, context)
 
-    def create_xlsx_report(self, data, objs):
+    def create_xlsx_report(self, ids, data, report):
+        self.parser_instance = self.parser(
+            self.env.cr, self.env.uid, self.name2, self.env.context)
+        objs = self.getObjects(
+            self.env.cr, self.env.uid, ids, self.env.context)
+        self.parser_instance.set_context(objs, data, objs.ids, 'xlsx')
```

Comment:
About the signature, it follows the standard pattern used by Odoo: https://github.com/OCA/OCB/blob/9.0/openerp/report/report_sxw.py#L371

Then, objects are instanciated inside the method thanks to the already defined `getObjects` method.

About the parser class, it is used to inject data or functions to the report. Having the parser instance in the method which generates the XLSX file allows to access these data/functions easily. The `report_xls` module send this parser instance as a parameter: https://github.com/OCA/reporting-engine/blob/8.0/report_xls/report_xls.py#L169 , maybe it's better to do it this way?

Another use case is to share one parser (which prepares the data) with 2 different report classes.